br/pkg/storage/s3.go

```diff
@@ -387,11 +392,13 @@ func newS3Storage(backend *backuppb.S3, opts *ExternalStorageOptions) (obj *S3St
 		}
 	}
 
-	return &S3Storage{
+	s3Storage := &S3Storage{
 		session: ses,
 		svc:     c,
 		options: &qs,
-	}, nil
+	}
+	qs.ObjectLockEnabled = s3Storage.checkObjectLockEnabled()
```

Comment:
Why not just fill the `ObjectLockEnabled` directly into `s3Storage.ObjectLockEnabledEnabled`?

Reply:
(Well, it should be sent to TiKV, however should we modify the `kvproto` before doing that?)